### PR TITLE
Fixed aside handler invocation to correctly set up runtime

### PR DESCRIPTION
**Pull request by gsidebo**
This PR changes the invocation of XBlockAside AJAX handler functions. It makes sure the runtime is correctly set up for the aside in both LMS and Studio

### CODE REVIEW

**Comment by gsidebo** [comment]
I updated my rapid response PR (https://github.com/mitodl/rapid-response-xblock/pull/18). These fixes can be tested with that branch

**Comment by noisecapella** [comment]
I'm seeing this error:

    edx.devstack.lms | Traceback (most recent call last):
    edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    edx.devstack.lms |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
    edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 145, in inner
    edx.devstack.lms |     return func(*args, **kwargs)
    edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/module_render.py", line 970, in handle_xblock_callback
    edx.devstack.lms |     return _invoke_xblock_handler(request, course_id, usage_id, handler, suffix, course=course)
    edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/module_render.py", line 1090, in _invoke_xblock_handler
    edx.devstack.lms |     request, course_id, unicode(block_usage_key), aside_usage_id=aside_usage_id, course=course
    edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/module_render.py", line 1042, in get_module_by_usage_id
    edx.devstack.lms |     course=course
    edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/module_render.py", line 389, in get_module_for_descriptor
    edx.devstack.lms |     course=course
    edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/module_render.py", line 838, in get_module_for_descriptor_internal
    edx.devstack.lms |     descriptor_to_bind.bind_for_student(
    edx.devstack.lms | AttributeError: 'RapidResponseAside' object has no attribute 'bind_for_student'
    edx.devstack.lms | [02/Mar/2018 12:18:38] "GET /courses/course-v1:ReplaceStatic+ReplaceStatic+2018_T1/xblock/aside-usage-v2:block-v1$:ReplaceStatic+ReplaceStatic+2018_T1+type@problem+block@b9dca79886ef481dbade98a50de54673::rapid_response_xblock/handler/toggle_block_open_status HTTP/1.1" 500 115538


This occurs when I'm viewing the LMS output as staff and click on the toggle button

**Comment by gsidebo** [comment]
Are you using my rapid response branch to test it, and is it up to date? If
so, it's a bug. I'll take a look

**Comment by noisecapella** [comment]
Yes and it was the latest at the time

**Comment by gsidebo** [comment]
@noisecapella OK try pulling from the rapid response branch and test again

**Comment by noisecapella** [comment]
I can confirm it works for me now, looking over the code...

File: cms/djangoapps/contentstore/views/component.py [hunk 1]
```diff
@@ -442,22 +441,29 @@ def component_handler(request, usage_key_string, handler, suffix=''):
...
     try:
-        descriptor = modulestore().get_item(usage_key)
-        descriptor.xmodule_runtime = StudioEditModuleRuntime(request.user)
-        resp = descriptor.handle(handler, req, suffix)
+        if is_xblock_aside(usage_key):
+            # Get the descriptor for the block being wrapped by the aside (not the aside itself)
+            descriptor = modulestore().get_item(usage_key.usage_key)
+            aside_instance = get_aside_from_xblock(descriptor, usage_key.aside_type)
+            aside_instance.runtime = StudioEditModuleRuntime(request.user)
+            asides = [aside_instance] if aside_instance else []
```
Comment (noisecapella) [inline cms/djangoapps/contentstore/views/component.py]: I'm not sure if you need the if clause here, since if it was `None` the previous line would fail

**Review by noisecapella — CHANGES_REQUESTED** [review]
Just the one comment, everything else looks good

**Review by noisecapella — APPROVED** [review]

**Pull request merged**